src/main/io/ledstrip.h

```diff
@@ -86,46 +123,42 @@ typedef struct specialColorIndexes_s {
     uint8_t color[LED_SPECIAL_COLOR_COUNT];
 } specialColorIndexes_t;
 
-typedef struct ledConfig_s {
-    uint8_t xy;     // see LED_X/Y_MASK defines
-    uint8_t color;  // see colors (config_master)
-    uint16_t flags; // see LED_FLAG_FUNCTION + LED_FLAG_DIRECTION
-} ledConfig_t;
+typedef uint32_t ledConfigPacked_t;
 
-#define LED_X_BIT_OFFSET 4
-#define LED_Y_BIT_OFFSET 0
-#define LED_XY_MASK      0x0F
+typedef struct ledCounts_s {
+    uint8_t ledCount;
+    uint8_t ledRingCount;
+    uint8_t ledLarsonScannerCount;
+    uint8_t ledRingSeqLen;
+} ledCounts_t;
 
-static inline int ledGetX(const ledConfig_t *lcfg) { return (lcfg->xy >> LED_X_BIT_OFFSET) & LED_XY_MASK; }
-static inline int ledGetY(const ledConfig_t *lcfg) { return (lcfg->xy >> LED_Y_BIT_OFFSET) & LED_XY_MASK; }
-static inline void ledSetXY(ledConfig_t *lcfg, int x, int y) {
-    lcfg->xy = ((x & LED_XY_MASK) << LED_X_BIT_OFFSET) | ((y & LED_XY_MASK) << LED_Y_BIT_OFFSET);
-}
-#define CALCULATE_LED_XY(x, y) ((((x) & LED_XY_MASK) << LED_X_BIT_OFFSET) | (((y) & LED_XY_MASK) << LED_Y_BIT_OFFSET))
+#define LED(x, y, col, dir, func, ol, params) LED_MOV_POS(CALCULATE_LED_XY(x, y)) | LED_MOV_COLOR(col) | LED_MOV_DIRECTION(dir) | LED_MOV_FUNCTION(func) | LED_MOV_OVERLAY(ol) | LED_MOV_PARAMS(params)
```

Comment:
parentheses are missing here, operator priority may break result when used in bad context